src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobRecord.cs

```diff
@@ -0,0 +1,88 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Core.Messages.Export;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Core.Features.Export
+{
+    /// <summary>
+    /// Class to hold metadata for an individual export request.
+    /// </summary>
+    public class ExportJobRecord
+    {
+        public ExportJobRecord(CreateExportRequest exportRequest, int jobSchemaVersion)
+        {
+            EnsureArg.IsNotNull(exportRequest, nameof(exportRequest));
+            EnsureArg.IsGt(jobSchemaVersion, 0, nameof(jobSchemaVersion));
+
+            Request = exportRequest;
+            JobSchemaVersion = jobSchemaVersion;
+
+            // Default values
+            JobStatus = OperationStatus.Queued;
+            Id = Guid.NewGuid().ToString();
+            QueuedTime = DateTimeOffset.Now;
+            LastModifiedTime = DateTimeOffset.Now;
+        }
+
+        [JsonConstructor]
+        protected ExportJobRecord()
+        {
+        }
+
+        [JsonProperty(JobRecordProperties.Request)]
+        public CreateExportRequest Request { get; private set; }
+
+        [JsonProperty(JobRecordProperties.Id)]
+        public string Id { get; private set; }
+
+        [JsonProperty(JobRecordProperties.JobHash)]
+        public string JobHash { get; private set; }
+
+        [JsonProperty(JobRecordProperties.JobQueuedTime)]
+        public DateTimeOffset QueuedTime { get; private set; }
+
+        [JsonProperty(JobRecordProperties.PartitonKey)]
+        public string PartitionKey { get; private set; } = OperationsConstants.ExportJobPartitionKey;
```

Comment:
don't need the setter since this is a read-only property.

Reply:
Also, this PartitionKey is Cosmos DB specific need and therefore should not be in the Core. You will need to create a wrapper class just like ResourceWrapper and wrap this class with that.

Reply:
Created a new `CosmosExportJobRecordWrapper` class for this.